confdb/confdb.go

```diff
@@ -1387,62 +1574,148 @@ func (p viewRule) isWriteable() bool {
 // pattern is an individual subkey of a dot-separated name or path pattern. It
 // can be a literal value of a placeholder delineated by curly brackets.
 type requestMatcher interface {
-	match(subkey string, placeholders map[string]string) bool
-	String() string
+	accessor
+
+	match(subkey string, matched *matchedPlaceholders) bool
+}
+
+type writeOpts struct {
+	topLevel bool
 }
 
 type storageWriter interface {
-	write(sb *strings.Builder, placeholders map[string]string) error
+	accessor
+
+	write(sb *strings.Builder, matched *matchedPlaceholders, opts writeOpts)
 }
 
 // placeholder represents a subkey of a name/path (e.g., "{foo}") that can match
 // with any value and map it from the input name to the path.
-type placeholder string
+type keyPlaceholder string
 
-// match adds a mapping to the placeholders map from this placeholder key to the
+// match adds an entry to matchedPlaceholders mapping this placeholder key to the
 // supplied name subkey and returns true (a placeholder matches with any value).
-func (p placeholder) match(subkey string, placeholders map[string]string) bool {
-	placeholders[string(p)] = subkey
+func (p keyPlaceholder) match(subkey string, matched *matchedPlaceholders) bool {
```

Comment:
will the match methods continue to take a string or use an accessor at some point?

Reply:
I think they'll use an accessor but we need the Get and Set changes in place first (since we can't change the interface for just one side)